server-spi-private/src/main/java/org/keycloak/credential/SingleUserCredentialManagerProviderFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2022. Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.credential;
+
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.provider.ProviderFactory;
+
+public interface SingleUserCredentialManagerProviderFactory extends ProviderFactory<SingleUserCredentialManagerProvider> {
```

Comment:
I tend to think that this interface should vanish

Reply:
Changed to a concrete implementation in the next commit and it is called LegacySingleUserCredentialManager. The accompanying LegacySingleUserCredentialManagerStrategy will eventually be folded into that class but not yet to make it simpler to compare with the MapSingleUserCredentialManager